store/cachekv/store.go

```diff
@@ -93,6 +93,32 @@ func (store *Store) Delete(key []byte) {
 	store.setCacheValue(key, nil, true)
 }
 
+type cEntry struct {
+	key string
+	val *cValue
+}
+
+func (store *Store) deleteCaches() {
+	if len(store.cache) > 100_000 {
+		// Cache is too large. We likely did something linear time
+		// (e.g. Epoch block, Genesis block, etc). Free the old caches from memory, and let them get re-allocated.
+		// TODO: In a future CacheKV redesign, such linear workloads should get into a different cache instantiation.
+		store.cache = make(map[string]*cValue)
+		store.unsortedCache = make(map[string]struct{})
+	} else {
+		// Clear the cache using the map clearing idiom
+		// and not allocating fresh objects.
+		// Please see https://bencher.orijtech.com/perfclinic/mapclearing/
+		for key := range store.cache {
+			delete(store.cache, key)
+		}
+		for key := range store.unsortedCache {
+			delete(store.unsortedCache, key)
+		}
+	}
```

Comment:
```go
if cond {
    x.f1 = make(...)
    x.f2 = make(...)
} else {
    for k := range x.f1 {
        delete(x.f1, k)
    }
    for k := range x.f2 {
        delete(x.f2, k)
    }
```

These two blocks of code are equivalent. There is no reason to differentiate between cond and not-cond.